layouts/blog/activityreport.html

```diff
@@ -0,0 +1,93 @@
+{{/* custom layout to report on project activity for a specified date range.
+
+Page metadata should use the following structure to configure dates:
+
+layout: activityreport
+date_range:
+    start: "2020-06-01"
+    end: "2020-08-01"
+*/}}
+{{ define "main" }}
+<div class="td-content">
+    <h1>{{ .Title }}</h1>
+    {{ with .Params.description }}<div class="lead">{{ . | markdownify }}</div>{{ end }}
+    <div class="td-byline mb-4">
+        {{ with .Params.author }}{{ T "post_byline_by" }} <b>{{ . | markdownify }}</b> |{{ end}}
+        <time datetime="{{  $.Date.Format "2006-01-02" }}" class="text-muted">{{ $.Date.Format $.Site.Params.time_format_blog  }}</time>
+    </div>
+
+{{ .Content }}
+
+<hr/>
+
+{{ $date_range := .Params.date_range }}
+
+{{/* gather iteration summary data for the specified date range */}}
+{{ $project_iterations := (where (where $.Site.Data.iteration_summary "from" ">=" $date_range.start) "to" "<=" $date_range.end)  }}
+
+<div id="project-points"></div>
+
+<div id="project-issues"></div>
+
+<script id="iteration-data" type="application/json">{{ $project_iterations|jsonify (dict "indent" "  ") | safeJS }}</script>
+<script id="project-colors" type="application/json">{{ $.Site.Data.project_colors|jsonify (dict "indent" "  ") | safeJS }}</script>
+
+<script>
+document.addEventListener('DOMContentLoaded', (event) => {
+    const iterationData = JSON.parse(document.getElementById('iteration-data').textContent);
+    const colors = JSON.parse(document.getElementById('project-colors').textContent);
+
+    // generate unique list of projects included in this data
+    let projects = iterationData.map(x => Object.keys(x.project_points))
+        .flat()
```

Comment:
you can save one call and a little time by using [`Array.prototype.flatMap()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) 🙂

Reply:
thank you